src/rev_ai/custom_vocabularies_client.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+"""Speech recognition tools for using Rev.ai"""
+
+from .baseclient import BaseClient
+from . import utils
+
+try:
+    from urllib.parse import urljoin
+except ImportError:
+    from urlparse import urljoin
+
+
+class RevAiCustomVocabulariesClient(BaseClient):
+    """Client which implements Rev.ai CustomVocabulary API"""
+
+    def __init__(self, access_token):
+        """Constructor
+
+        :param access_token: access token which authorizes all requests and
+                             links them to your account. Generated on the
+                             settings page of your account dashboard
+                             on Rev.ai
+        """
+        BaseClient.__init__(self, access_token)
+
+        self.base_url = urljoin(self.base_url, 'vocabularies/')
+
+    def submit_custom_vocabularies(
+            self,
+            custom_vocabularies,
+            callback_url=None,
+            metadata=None):
+        """Submit custom vocabularies.
+
+        :param custom_vocabularies: List of CustomVocabulary objects
+        :param callback_url: callback url to invoke on job completion as a
+                             webhook
+        :param metadata: info to associate with the transcription job
+        """
+
+        if not custom_vocabularies:
+            raise ValueError('custom_vocabularies must be provided')
+
+        payload = self._create_custom_vocabularies_options_payload(
+            custom_vocabularies,
+            callback_url,
+            metadata
+        )
+
+        response = self._make_http_request(
```

Comment:
Is it possible to create a mock of the base client and assert that this function was called with the correct parameters?

We are missing test logic for all function calls in both this cv client and the api client